client/src/components/ModalLogIn.tsx

```diff
@@ -1,23 +1,60 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
 import { Formik, FormikValues } from 'formik';
-import { Dispatch, SetStateAction } from 'react';
-import { Modal, Pressable, StyleSheet, Text, TextInput, View } from 'react-native';
+import { Dispatch, SetStateAction, useState } from 'react';
+import {
+  Modal,
+  Pressable,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import * as Yup from 'yup';
+import { StackParamList } from '../../App';
+import { IUser } from '../constants/interfaces';
+import { signIn } from '../helpers/api';
 import { COLORS } from '../styles/colors';
 import { STYLES } from '../styles/styles';
 
 type ModalLogInProps = {
+  params: NativeStackScreenProps<StackParamList, 'Intro'>;
+  setUser: React.Dispatch<React.SetStateAction<IUser>>;
   modalLogInVisible: boolean;
   setModalLogInVisible: Dispatch<SetStateAction<boolean>>;
-}
+};
 
 const initialValues = {
   email: '',
   password: '',
 };
 
-export default function ModalLogIn({modalLogInVisible, setModalLogInVisible}: ModalLogInProps) {
-  const handleFormSubmit = (values: FormikValues) => {
-    console.log(values);
+const ValidationSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Required'),
+  password: Yup.string().required('Required'),
+});
+
+export default function ModalLogIn({
+  params,
+  setUser,
+  modalLogInVisible,
+  setModalLogInVisible,
+}: ModalLogInProps) {
+  const [error, setError] = useState('');
+
+  const handleFormSubmit = async (values: FormikValues) => {
+    try {
+      const user = await signIn(values.email, values.password);
+      console.log(user);
+      setModalLogInVisible(!modalLogInVisible);
+      setUser(user);
+      params.navigation.navigate('Home');
```

Comment:
In general it's considered a good practice to create enum or dictionary with pages, so you could refer them via 1 key, also IDE will tell you what page you have there